CMakeLists.txt

```diff
@@ -1,6 +1,13 @@
 cmake_minimum_required(VERSION 3.4)
 
-project(LibVNCServer VERSION 0.9.14 LANGUAGES C)
+set(PROJECT_LANGUAGES C)
+
+if(NOT ${CMAKE_CXX_COMPILER} STREQUAL "")
+  set(CMAKE_CXX_STANDARD 17)
+  set(PROJECT_LANGUAGES C CXX)
+endif(NOT ${CMAKE_CXX_COMPILER} STREQUAL "")
```

Comment:
Newer CMake does not require this repetition

Reply:
Agreed, but since the original CMakeLists follows this pattern, I chose to follow it as well.